activesupport/lib/active_support/callbacks.rb

```diff
@@ -787,10 +787,13 @@ def set_callback(name, *filter_list, &block)
           end
         end
 
-        # Skip a previously set callback. Like +set_callback+, <tt>:if</tt> or
+        # Skip a previously set callback. Like #set_callback, <tt>:if</tt> or
         # <tt>:unless</tt> options may be passed in order to control when the
         # callback is skipped.
         #
+        # Note: this example uses +PersonRecord+ and +#saving_message+, which you
+        # can see defined here[rdoc-ref:ActiveSupport::Callbacks]
+        #
```

Comment:
It would be nice if all `ClassMethods` modules are included as class methods on the class itself, instead of showing them on a seperate page. But I'm not sure how easy it is to implement that in SDoc.

Maybe it makes sense to define PersonRecord here as well, and use a single class:

```ruby
class PersonRecord
  include ActiveSupport::Callbacks
  define_callbacks :save
  set_callback :save, :before, :saving_message
  def saving_message
    puts "saving..."
  end

  def save
    run_callbacks :save do
      puts "- save"
    end
  end
end
```

Reply:
Thanks for the review! I kind of agree I don't really like linking here, although it is unfortunate that the example would be duplicated. I haven't done much digging yet but I wonder how often example setup is shared between method documentation or if its relatively infrequent.

(Also agree that `ClassMethods` feels like it should be on the `Callbacks` doc itself, which I think would make the shared example less confusing)

Reply:
The existing PersonRecord is also difficult to reason about as it is a subclass of Record. It's a lot of code to review.

I'm also okay with merging it for now, as this is better than the current documentation.

Reply:
That works for me, I'll merge and then work on replacing this with a more condensed example